.github/workflows/build-firmware-and-emulator.yml

```diff
@@ -14,58 +14,125 @@ env:
 
 jobs:
   Build-Firmware-And-Emulator:
-    runs-on: windows-latest
+    strategy:
+      matrix:
+        os: [ windows, osx-arm, osx-intel ]
```

Comment:
Wanna throw a Linux build in here for kicks?

Reply:
Sidebar, Mac builds are reasonably speedy. Nice. msys2 and the IDF are killers. We should probably be [doing some caching](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows).

Reply:
> Wanna throw a Linux build in here for kicks?

Ah yeah, might as well!

Reply:
> Sidebar, Mac builds are reasonably speedy. Nice. msys2 and the IDF are killers. We should probably be [doing some caching](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows).

Hmm, I took a look but it looks like all those are just for language-level package managers like node and stuff. But! I did find this action which claims to support caching, I'll give it a try: https://github.com/msys2/setup-msys2

Reply:
Nice, Linux builds in under 30 seconds! :sunglasses:

I tried the msys2 setup stuff, but it does a bunch of extra stuff I don't really understand and fails. I'll try changing the `MSYSTEM` config and see if that helps but if it fails again I'll just leave it for later.

Reply:
Okay, just caching the IDF knocks about 5 minutes off the Windows build so now it's done in a _blazing_ ten minutes!